### ISSUE TITLE
truncate coordinate dimensions of Layer/Feature/Geometry

### ISSUE THREAD

**Issue by jomel** [naturalatlas/node-gdal#263]
I'd like to truncate coordinate dimensions of all features in a dataset to two (XY) - the equivalent of running `ogr2ogr -f GML output_2d.gml input_3d.gml -dim 2`

I can see that [Geometry class](http://naturalatlas.github.io/node-gdal/classes/gdal.Geometry.html?#method-isSimple) has dimension property, but that is read-only.

Is there a method at a dataset, layer, feature, or geometry level that I can use?

**Comment by brandonreavis** [comment]
When you are creating a new layer in the output dataset (http://naturalatlas.github.io/node-gdal/classes/gdal.DatasetLayers.html#method-create) use a 2D geometry type instead of a 2.5D type (ie. `gdal.wkbPoint` vs `gdal.wkbPoint25D`).

Here is an example of how to go about it:

```js
var newDataset = gdal.open('output_2d.gml', 'w', 'GML');
var newLayer = newDataset.layers.create('output_2d', null, sourceLayer.geomType & (~gdal.wkb25DBit))
sourceLayer.fields.forEach(fieldDefn => {
  newLayer.fields.add(new gdal.FieldDefn(fieldDefn.name, fieldDefn.type));
});
sourceLayer.features.forEach(feature => {
  var newFeature = new gdal.Feature(newLayer);
  newFeature.setGeometry(feature.getGeometry());
  newFeature.fields.set(feature.fields.toObject());
  newLayer.features.add(newFeature);
});
newDataset.close();
```

If this still retains the elevation data, you'll have to create a new geometry object and copy all the points over to it and just omit the z coordinate.

**Comment by jomel** [comment]
Thanks for the advice, but it seems that  `newFeature.setGeometry(feature.getGeometry());` dosn't respect the geometry definitions set in the layer creation, and I am still left with 3-dimension coordinates.
I've written a custom function to trim each coordinate set in turn, but it feels a silly waste of processing time, since node-gdal is supposed to be a replacement for ogr2ogr in node, and it does not support this.

**Comment by brandonreavis** [comment]
Ahh you're right. I didn't realize gdal had a setCoordinateDimension method on geometry objects for this. Once the builds finish running, this will be available in 0.9.9. You'll be able to use it like:

``` js
var geometry = feature.getGeometry();
geometry.coordinateDimension = 2;
newFeature.setGeometry(geometry)
```
